daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/processors/parsers/DelimiterParsers.scala

```diff
@@ -73,6 +76,35 @@ class DelimiterTextParser(
     foundLocalDFAIndex >= 0
   }
 
+  private def localDelimiters(state: PState): Seq[DFADelimiter] = {
+    var foundLocalDFAIndex = -1
+    val localIndexStart = state.mpstate.delimitersLocalIndexStack.top
+    val inScopeDelimiters = state.mpstate.delimiters
+    val res = inScopeDelimiters.slice(localIndexStart, inScopeDelimiters.length)
+    res
+  }
+
+  private def didNotFindExpectedDelimiter(foundDelimiter: ParseResult, start: PState): Unit = {
+    val delimString = foundDelimiter.matchedDelimiterValue.get
+    val localDelims = localDelimiters(start)
+    val foundDFA = foundDelimiter.matchedDFAs(0)
+    PE(
+      start,
+      s"""Found enclosing delimiter: %s during scan for local delimiter(s): %s.
+         | The expected delimiter(s) were: %s.
+         | The enclosing delimiter was from %s.
+         |""".stripMargin,
+      foundDFA.strForDiagnostic,
+      localDelims.map { d => d.strForDiagnostic }.mkString(", "),
+      localDelims
```

Comment:
This duplicates localDelims. THoughts on an error message like

```scala
PE(
  "Found enclosing delimiter %s from %s.
  |Expected delimiter(s) were:
  |%s",
  foundDFA.strForDiagnostic,
  foundDFA.locationDescription,
  localDelims.map { d =>
    s"  ${d.delimType.toString} ${d.strForDiagnostic} from location ${d.location}."
  }
)
```

Reply:
I was thinking getting what was not found and what was found instead onto the first line of the message would be helpful with other details on later lines.